overlord/snapstate/component.go

```diff
@@ -222,17 +391,23 @@ func doInstallComponent(st *state.State, snapst *SnapState, compSetup ComponentS
 	}
 
 	// security
-	if !compSetup.JointSnapComponentsInstall {
-		setupSecurity := st.NewTask("setup-profiles", fmt.Sprintf(i18n.G("Setup component %q%s security profiles"), compSi.Component, revisionStr))
+	if !compSetup.MultiComponentInstall && setupSecurity == nil {
+		setupSecurity = st.NewTask("setup-profiles", fmt.Sprintf(i18n.G("Setup component %q%s security profiles"), compSi.Component, revisionStr))
+		setupSecurity.Set("component-setup-task", prepare.ID())
+		setupSecurity.Set("snap-setup-task", snapSetupTaskID)
 		componentTS.beforeLink = append(componentTS.beforeLink, setupSecurity)
-		addTask(setupSecurity)
+	}
+
+	if setupSecurity != nil {
+		setupSecurity.WaitFor(prev)
```

Comment:
probably worth an explicit comment about not using addTask here because the task is shared